### PR TITLE
`Tutorial groups`: Fix an issue in an authentication check

### PR DESCRIPTION
**Pull request by stefanwaldhauser**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.

### Motivation and Context
- There was a bug in the AuthenticationCheckService when course or teaching assistant were null for a tutorial group because of hiding these properties for privacy reasons

### Description

- Always load the tutorial group from database in the authentication check service to prevent this error.

### Steps for Testing
Nothing to test really. Bug and bugfix is clear from the code

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage
No changes

### CODE REVIEW

**Review by manuelmanso — DISMISSED** [review]
Fix looks good

**Review by Strohgelaender — DISMISSED** [review]
code looks good

**Review by manuelmanso — APPROVED** [review]
Re-approve

File: src/test/java/de/tum/in/www1/artemis/tutorialgroups/TutorialGroupIntegrationTest.java [hunk 1]
```diff
@@ -643,18 +658,44 @@ private void verifyPrivateInformationIsShown(TutorialGroup tutorialGroup, Intege
...
+        TutorialGroup tutorialGroupsOfCourse;
+        if (loadFromService) {
+            var user = userRepository.findOneByLogin(userLogin).get();
+            var course = courseRepository.findById(exampleCourseId).get();
+            tutorialGroupsOfCourse = tutorialGroupService.getOneOfCourse(course, user, exampleOneTutorialGroupId);
+        }
+        else {
+            tutorialGroupsOfCourse = request.get("/api/courses/" + exampleCourseId + "/tutorial-groups/" + exampleOneTutorialGroupId, HttpStatus.OK, TutorialGroup.class);
+        }
+        return tutorialGroupsOfCourse;
```
Comment (Strohgelaender) [inline src/test/java/de/tum/in/www1/artemis/tutorialgroups/TutorialGroupIntegrationTest.java:683]:
```suggestion
        if (loadFromService) {
            var user = userRepository.findOneByLogin(userLogin).get();
            var course = courseRepository.findById(exampleCourseId).get();
            return tutorialGroupService.getOneOfCourse(course, user, exampleOneTutorialGroupId);
        }
        else {
            return request.get("/api/courses/" + exampleCourseId + "/tutorial-groups/" + exampleOneTutorialGroupId, HttpStatus.OK, TutorialGroup.class);
        }
```

File: src/test/java/de/tum/in/www1/artemis/tutorialgroups/TutorialGroupIntegrationTest.java [hunk 2]
```diff
@@ -643,18 +658,44 @@ private void verifyPrivateInformationIsShown(TutorialGroup tutorialGroup, Intege
...
+        if (loadFromService) {
+            var user = userRepository.findOneByLogin(userLogin).get();
+            var course = courseRepository.findById(exampleCourseId).get();
+            tutorialGroupsOfCourse = tutorialGroupService.findAllForCourse(course, user).stream().toList();
+        }
+        else {
+            tutorialGroupsOfCourse = request.getList("/api/courses/" + exampleCourseId + "/tutorial-groups", HttpStatus.OK, TutorialGroup.class);
+        }
+        return tutorialGroupsOfCourse;
+    }
```
Comment (Strohgelaender) [inline src/test/java/de/tum/in/www1/artemis/tutorialgroups/TutorialGroupIntegrationTest.java:697]:
```suggestion
    private List<TutorialGroup> getTutorialGroupsOfExampleCourse(boolean loadFromService, String userLogin) throws Exception {
        if (loadFromService) {
            var user = userRepository.findOneByLogin(userLogin).get();
            var course = courseRepository.findById(exampleCourseId).get();
            return tutorialGroupService.findAllForCourse(course, user).stream().toList();
        }
        else {
            return request.getList("/api/courses/" + exampleCourseId + "/tutorial-groups", HttpStatus.OK, TutorialGroup.class);
        }
    }
```

File: src/main/java/de/tum/in/www1/artemis/service/CourseService.java [hunk 3]
```diff
@@ -228,11 +228,6 @@ public Course findOneWithExercisesAndLecturesAndExamsAndLearningGoalsAndTutorial
         if (authCheckService.isOnlyStudentInCourse(course, user)) {
             course.setExams(examRepository.filterVisibleExams(course.getExams()));
         }
-        course.getTutorialGroups().forEach(tutorialGroup -> {
```
Comment (stefanwaldhauser) [inline src/main/java/de/tum/in/www1/artemis/service/CourseService.java:231]: I removed this check as it is already done in `findAllForCourse` above.

**Review by Strohgelaender — APPROVED** [review]

**Review by manuelmanso — APPROVED** [review]

**Pull request merged**